Foundation/CGFloat.swift

```diff
@@ -949,3 +949,35 @@ extension CGFloat : _CVarArgPassedAsDouble, _CVarArgAligned {
         return native._cVarArgAlignment
     }
 }
+
+extension CGFloat : Codable {
+    @_transparent
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.singleValueContainer()
+        do {
+            self.native = try container.decode(NativeType.self)
+        } catch DecodingError.typeMismatch(let type, let context) {
+            // We may have encoded as a different type on a different platform. A
+            // strict fixed-format decoder may disallow a conversion, so let's try the
+            // other type.
+            do {
+                if NativeType.self == Float.self {
+                    self.native = NativeType(try container.decode(Float.self))
```

Comment:
Philippe is correct. This is a fallback case for encoders which differentiate between `Double`s and `Float`s in a way that makes them incompatible. On a platform where `CGFloat.NativeType == Double.self`, we will encode it as a `Double`. If we try to read it back on a platform where `CGFloat.NativeType == Float.self`, however, the decode could fail since the `Double` value would not fit in a `Float`. So this is that fallback case — if our native type is a `Float` but we failed to read it as a `Float`, try to read it as a `Double` since the decode may have failed.

The code is correct — you'll want to have the types the same way here as in the Foundation overlay.